sdks/java/io/bigquery-io-tests/src/test/java/org/apache/beam/sdk/bigqueryiotests/BigQueryIOIT.java

```diff
@@ -113,33 +120,34 @@ public static void setup() throws IOException {
         SyntheticOptions.fromJsonString(options.getSourceOptions(), SyntheticSourceOptions.class);
     bigQueryMetricsDataset = options.getBigQueryDataset();
     bigQueryMetricsTable = options.getBigQueryTable();
-    bigQuerySourceDataset = options.getBigQuerySourceDataset();
-    bigQuerySourceTable = options.getBigQuerySourceTable();
+    bigQueryTestDataset = options.getBigQueryTestDataset();
+    bigQueryTestTable = options.getBigQueryTestTable();
+    useStreamingWrites = options.getUseStreamingWrites();
     BigQueryOptions bigQueryOptions = BigQueryOptions.newBuilder().build();
     tableQualifier =
         String.format(
-            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
-    if (!checkForSourceTable()) {
-      createAndFillSourceTable();
-    }
+            "%s:%s.%s", bigQueryOptions.getProjectId(), bigQueryTestDataset, bigQueryTestTable);
   }
 
   @AfterClass
   public static void tearDown() {
     BigQueryOptions options = BigQueryOptions.newBuilder().build();
     BigQuery client = options.getService();
-    TableId tableId =
-        TableId.of(options.getProjectId(), bigQuerySourceDataset, bigQuerySourceTable);
+    TableId tableId = TableId.of(options.getProjectId(), bigQueryTestDataset, bigQueryTestTable);
     client.delete(tableId);
   }
 
-  private static void createAndFillSourceTable() {
-
-    Pipeline pipeline = Pipeline.create();
+  private void testWrite() {
+    Pipeline pipeline = Pipeline.create(options);
 
+    BigQueryIO.Write.Method method =
+        useStreamingWrites
```

Comment:
This will be easier and there will be no branches if you use the enum :)

Reply:
You're right, fixed!